cluster_management/src/main/java/com/pinterest/rocksplicator/Spectator.java

```diff
@@ -129,6 +131,20 @@ public static void main(String[] args) throws Exception {
     Spectator spectator= new Spectator(zkConnectString, clusterName, instanceName);
 
     CuratorFramework zkClient = CuratorFrameworkFactory.newClient(zkConnectString, new ExponentialBackoffRetry(1000, 3));
+
+    zkClient.getConnectionStateListenable().addListener(new ConnectionStateListener() {
+      @Override
+      public void stateChanged(CuratorFramework client, ConnectionState newState) {
+      // If the connection state changes to LOST, then it means it already lost the lock and
+      // the current leader role is not legal any more. Here we simply terminate it and rely
+      // on restarting to re-acquire the inter-process lock.
+        if (newState == ConnectionState.LOST) {
```

Comment:
according to the errorhandling section on http://curator.apache.org/curator-recipes/shared-lock.html We should commit suicide for SUSPENDED event as well.

Can we remove the TODO from line 153 to line 164 below? As it's not the case with this PR.

Reply:
I was thinking the SUSPENDED status will turn into RECONNECTED or LOST eventually, and only need to handle the SUSPENDED->LOST case where the lock is released. And you are right, to make it safe and clean, we should also commit suicide in SUSPENDED evnet.